internal/api/handlers.go

```diff
@@ -114,13 +120,49 @@ func GetRecommendationSet(c echo.Context) error {
 	XRHID := c.Get("Identity").(identity.XRHID)
 	OrgID := XRHID.Identity.OrgID
 	user_permissions := get_user_permissions(c)
+	handlerName := "recommendationset"
 
 	RecommendationIDStr := c.Param("recommendation-id")
 	RecommendationUUID, err := uuid.Parse(RecommendationIDStr)
 	if err != nil {
 		return c.JSON(http.StatusBadRequest, echo.Map{"status": "error", "message": "bad recommendation_id"})
 	}
 
+	var unitChoices = make(map[string]string)
+
+	cpuUnitParam := c.QueryParam("cpu-unit")
+	var cpuUnitOptions = map[string]bool{
+		"millicores": true,
+		"cores":      true,
+	}
+
+	if cpuUnitParam != "" {
+		if !cpuUnitOptions[cpuUnitParam] {
+			return c.JSON(http.StatusBadRequest, echo.Map{"status": "error", "message": "invalid cpu unit"})
+		} else {
+			unitChoices["cpu"] = cpuUnitParam
+		}
+	} else {
+		unitChoices["cpu"] = "cores"
+	}
+
+	memoryUnitParam := c.QueryParam("memory-unit")
+	var memoryUnitOptions = map[string]bool{
+		"bytes": true,
+		"MiB":   true,
+		"GiB":   true,
+	}
+
+	if memoryUnitParam != "" {
+		if !memoryUnitOptions[memoryUnitParam] {
+			return c.JSON(http.StatusBadRequest, echo.Map{"status": "error", "message": "invalid memory unit"})
+		} else {
+			unitChoices["memory"] = memoryUnitParam
+		}
+	} else {
+		unitChoices["memory"] = "MiB"
+	}
```

Comment:
Not a blocker, maybe we can create a common func for memory and cpu.

Reply:
each does things a bit differently from another, for instance cpu values require truncation whereas memory doesn't, hence the separation

Reply:
Apologies, I should have highlighted the block! I mean below blocks look identical,
~~~
	if cpuUnitParam != "" {
		if !cpuUnitOptions[cpuUnitParam] {
			return c.JSON(http.StatusBadRequest, echo.Map{"status": "error", "message": "invalid cpu unit"})
		} else {
			unitChoices["cpu"] = cpuUnitParam
		}
	} else {
		unitChoices["cpu"] = "cores"
	}

	if memoryUnitParam != "" {
		if !memoryUnitOptions[memoryUnitParam] {
			return c.JSON(http.StatusBadRequest, echo.Map{"status": "error", "message": "invalid memory unit"})
		} else {
			unitChoices["memory"] = memoryUnitParam
		}
	} else {
		unitChoices["memory"] = "MiB"
	}
~~~